packages/gitbook/src/app/(site)/fetch.ts

```diff
@@ -88,20 +104,26 @@ async function resolvePage(
         return page;
     }
 
-    // If page can't be found, we try with the API, in case we have a redirect
-    // We use the raw pathname to handle special/malformed redirects setup by users in the GitSync.
-    // The page rendering will take care of redirecting to a normalized pathname.
-    //
     // We don't test path that are too long as GitBook doesn't support them and will return a 404 anyway.
     if (rawPathname.length <= 512) {
-        const resolved = await getRevisionPageByPath(
-            contentTarget.spaceId,
-            contentTarget.revisionId,
-            rawPathname,
-        );
+        // If page can't be found, we try with the API, in case we have a redirect at space level.
+        // We use the raw pathname to handle special/malformed redirects setup by users in the GitSync.
+        // The page rendering will take care of redirecting to a normalized pathname.
+        const resolved = await getRevisionPageByPath(spaceId, revisionId, rawPathname);
         if (resolved) {
             return resolvePageId(pages, resolved.id);
         }
+
+        // If a page still can't be found, we try with the API, in case we have a redirect at site level.
+        const resolvedSiteRedirect = await getSiteRedirectBySource({
+            organizationId,
+            siteId,
+            source: rawPathname.startsWith('/') ? rawPathname : `/${rawPathname}`,
+            siteShareKey: input.shareKey,
+        });
+        if (resolvedSiteRedirect) {
+            return redirect(resolvedSiteRedirect.target);
```

Comment:
Using this [redirect](https://nextjs.org/docs/app/building-your-application/routing/redirecting#redirect-function) function returns a `307` which is fine for now but we'll need to think if we let users configure custom status codes for their redirects

Reply:
We should keep everything as temporary redirect for now.

Reply:
It's not a 307, it's a redirect done client-side. Same issue as 404.  https://github.com/vercel/next.js/blob/5b7833e3da2b22e45e5b13c7001d4d728d093133/packages/next/src/client/components/redirect-boundary.tsx#L28-L37